hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java

```diff
@@ -290,7 +290,12 @@ public ScmInfo getScmInfo() throws IOException {
       NodeManager nodeManager = scm.getScmNodeManager();
       Node client = nodeManager.getNode(clientMachine);
       List<Node> nodeList = new ArrayList();
-      nodes.stream().forEach(path -> nodeList.add(nodeManager.getNode(path)));
+      nodes.stream().forEach(path -> {
+        DatanodeDetails node = nodeManager.getNode(path);
+        if (node != null) {
```

Comment:
Can you elaborate the case nodeManager.getNode() gives back null? Can we add a WARN log for the case?

Reply:
nodeManager.getNode will return null when it can't find the node in the network topology or the node found is not a leaf node.  The WARN log for all these cases are  in nodeManager.getNode function.  

    if (node != null) {
      if (node instanceof InnerNode) {
        LOG.warn("Get node for {} return {}, it's an inner node, " +
            "not a datanode", address, node.getNetworkFullPath());
      } else {
        LOG.debug("Get node for {} return {}", address,
            node.getNetworkFullPath());
        return (DatanodeDetails)node;
      }
    } else {
      LOG.warn("Cannot find node for {}", address);
    }

Reply:
Thanks for the details. Makes sense to me.